### PR TITLE
Articulation Point in JAVA

### PR DESCRIPTION
**Pull request by krishna-y2000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description

I have contributed  Articulation Point problem of graph in  JAVA.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2491

### CODE REVIEW

File: Java/graphs/Articulation_Point.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*Problem Description
+A vertex is said to be an articulation point in a graph if removal of the vertex and associated edges disconnects the graph.
+You are given the nodes and edges . Find the articulation point.
+*/
+
+// Solution 
+import java.util.*;
+import java.lang.*;
+import java.io.*;
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java:8]: Add a new line after header ends.

File: Java/graphs/Articulation_Point.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+public class Articulation_Point {
+    // Initialize all the required variables with their type. 
+	private static int[] low =  new int[10001];
+	private static int[] visited = new int[10001]; 
+	private static int[] disc = new int[10001] ;
+	private static int timer = 1;
+	private static Set<Integer> hashset ;
+    static ArrayList<Integer> adj[];
+//  This class will help us to create a graph and know the articulation point.
+	static class Graph
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	 {
+		 static int v ;		
+        // Creating graph 
+		 Graph(int v)
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+		 {
+			 Graph.v = v;
+			 adj = new ArrayList[v+1];			 
+			 for(int i = 0 ; i < v+1 ; i++)
+			 {
+				 adj[i] = new ArrayList<Integer>();
+			 }
+		 }		 
+         // adding nodes with each other 
+		 static void addEdge(int u , int v)
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+		 }
+        // The method will help us to know the articulation point
+        /*
+            Code flow : 
+            step 1 :  We need to calculate the lowest discovery number. 
+            step 2 : The algorithm then searches for its adjacent vertices.
+            step 3 : the algorithms take a pair of vertices and checks whether its an articulation point or not.
+            step 4 : There’s one special case when the root is an articulation point
+        */    
+		 static void dfs(int node , int par)
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+						}
+						low[node] = Math.min(low[node] , low[child] );
+					}
+					else
+					{
+						low[node] = Math.min(low[node], disc[child] );
+					}									
+				}
+                // if root is discovery number
+				if(par == 0  && children > 1)
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+					if(visited[i] == 0 )
+						g.dfs(i , -1 );
+				}		
+			// Print all the articulation point
+				for(Integer val : hashset)
+				{
+					System.out.print(val + " ");
+				}	 
+		}
+  }
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Align your code and comments properly.

File: Java/graphs/Articulation_Point.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+4
+Enter number of edges
+5
+Enter values of edges
+1 0
+0 2
+2 1
+0 3
+3 4
+Output
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java:135]: Add a new line above output

File: Java/graphs/Articulation_Point.java [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+0 3
+Explanation :
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java:138]: Add a new line above explanation

File: Java/graphs/Articulation_Point.java [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+From all the nodes we find that articulation point is 0 and 3.This will seperate the graph and can make many unconnect graph
+Time Complexity : O(V + E)
+Space Complexity : O(V + E)
+*/
```
Comment (ankitaggarwal23) [inline Java/graphs/Articulation_Point.java]: Add a new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by krishna-y2000** [comment]
@ankitaggarwal23  changes are applied.

**Comment by krishna-y2000** [comment]
@ankitaggarwal23 hii,
if you find any spare time then please review my pr. This might be the last check.
@HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**